activesupport/lib/active_support/core_ext/file/atomic.rb

```diff
@@ -64,6 +64,8 @@ def self.probe_stat_in(dir) # :nodoc:
     file_name = join(dir, basename)
     FileUtils.touch(file_name)
```

Comment:
can it check if there is a directory instead? `FileUtils.touch(file_name) if File.directory?(dir)`

Reply:
It would still be subject to a small race condition:

```ruby
if Dir.exists?(dir)
  # other process delete the directory
  FileUtils.touch(file_name)
end
```

Rescueing the exception is better.

Reply:
exactly